### DISCUSSION TITLE
"Magic Spells"

### DISCUSSION THREAD

**Discussion by littlesera** [littlesera/LSCG#265] [Ideas]
"Spells" could be cast using a wand, or crafted into potions.
Maybe have a 'spellbook' that can build out as spells are 'learned'/'unlocked'?
Spell effects possibly countered by 'saving throws' like d&d?

Spell ideas:
- [x] 'stolen' orgasms (paired with someone else)
- [x] paired arousal (maybe the same as above)
- [x] outfit application (slime bomb, magic belts, etc)
- [x] mind control
- [x] sleep
- [x] heightened arousal
- [x] Blindness
- [x] Deafness
- [x] Silence/Gagged
- [x] Hold Person/Petrify
- [x] Enlarge/Reduce
- [x] Bless/Bane
- [x] Polymorph

**Comment by jimmyG1** [comment]
i love this

**Comment by Dwfreegethub** [comment]
I really like this idea, although I am not sure how you would learn the spells unless it is after casting so many of the lesser spells.  I would say it would also be fun to have a chance of failure or even a chance of affecting the caster.

And depending on how far you want to take it, deafen, mute, blindness, hold person would be other possibilities for spells.

If you do this is there a way to do a cast spell activity and then a new menu to choose the spell?

**Comment by anchillabc** [comment]
While I like the idea initially, I can't say the spell ideas themselves appeal much to me, and there are some I'd actively like to block. If implemented, needs to have a way to define what you're fine being hit with, with perhaps a "abc hits xyz with a spell but it fizzles" message to just indicate failure without giving more away.

**Comment by littlesera** [comment]
All great feedback, thank you!
I'm doing some refactors for the next little while to make this sort of work easier.
Needless to say, such a broad thing as "magic" will undoubtedly have some pretty extensive configuration options + some extra thinking around if spells need the target to also be running the mod. (both mechanically and for consent)

**Comment by littlesera** [comment]
Work has progressed on this in dev:
* Must be wielding a **wand** to "cast spells"
* "Spells" are **craftable**, and can combine up to three distinct effects
* A spell can be crafted into a **potion** using similar rules as drugged drinks if allowed. Paired effects are exempt from this.
* Each distinct spell effect can be **blocked** on the recipient end, and all magic is disabled by default.
  * Because of the above (and LSCG's opt-in philosophy) 'magic' will only work on other LSCG users who have enabled it on their end
* Allows for '**wild magic**' option, which at a basic level picks one of your spells at random, with a small chance to cast a truly random spell
  * Can be configured to always cast a truly random spell (1-3 distinct effects chosen at random)
  * Can be set to force wild magic (disable normal casting)
* Base magic settings can be remotely controlled with similar remote settings as hypnosis
* Spells can be 'taught' to other LSCG users with magic enabled, copying the spell definition into their known spells
  * Both characters must be holding a **wand**
  
  
Remaining...
* Allow resisting spell cast? How to handle that in UI?

**Reply by anchillabc** [reply]
My thoughts:
- Consider a chaotic failure similar to net gun (also with a toggle) with the spell effect applying to caster
- Don't handle spell resistance in the UI at all, handle it like other conflict issues like forced sipping etc. - it works well enough to reuse. Just put up a nice few messages like "<target> manages to dodge <caster>s spell." 
- Consider accepting previously crafted items like filled glasses and respirators with the same effect as spell sources as well to reduce the need for additional craft slot use (I recently reached the limit of allowed server storage due to MBS & FBC, don't additional crafts past the 4th page also count towards that via MBS?)
- Maybe elaborate on the rationale behind remote control of magic settings via remote control, is this perhaps as part of hypnosis? I cannot immediately think of a reason why.

**Comment by Dwfreegethub** [comment]
Another thought I came up with while watching people trying to take each other's wands.  Can it be an option to tie the wand to a person?

So for example name your wand "Wizard Sera's Wand" and only Sera could use it. Or at least successfully?  Maybe if someone else used it would backfire, 

Could also scan the description for the same.  If you did this for the wands it could just as easily be implemented for the guns as well, but it makes more sense for me for the wands.

**Comment by anchillabc** [comment]
Small thought for: Much like the comedic effect for netguns, consider allowing casting with vibrating wand and small vibe wand perhaps?

**Comment by Dwfreegethub** [comment]
I see in the Beta of LSCG 3.0 that you can still kneel and stand when Petrified.  In my mind, you would want that spell to keep them from moving at all.  I am not sure if you would want to freeze them to the point they cannot be moved at all like stone, or if they can be adjusted more like paralyzed.

**Reply by littlesera** [reply]
Good point. Noticed the full character access block for petrify and sleep is also overkill as it is also blocking safeword and room admin actions, which is unintended. Updated to block all pose changes on self.

**Reply by littlesera** [reply]
addressed in #284 284

**Comment by Dwfreegethub** [comment]
More thoughts on the Magic?  Is there, or have you thought about a way to selectively remove a spell and leave the rest in place?

**Reply by littlesera** [reply]
binary effects will cancel each other (enlarge/reduce, bless/bane). Everything else are all dispelled at once. Specific selection is probably a good feature for magic v2

**Comment by Dwfreegethub** [comment]
I hope I am not being too annoying with suggestions, but would there be a way to add an optional spell points, mana, or spell slots you can memorize at once? Anything like that which could add to the D&d feeling. Right now I can cast a spell at some one 10 times until it works. If I had a limited amount of spells I would think about it more before casting. Just a thought.

Also noticed layering is acting strange with the spells, but that might just be BC.

**Reply by littlesera** [reply]
Definitely, this is going to be the priority of the v.2 magic work, in addition to handling more edge cases around the effects as they are discovered.
Speaking of edge case, could you elaborate what you mean by layering issues?

**Comment by littlesera** [comment]
Extended to #299
